### PR TITLE
Adds debug arg to the CLI

### PR DESCRIPTION
**Pull request by sr-gi**
Debug logs can be turned in using `--debug`.

Also fixes pending logs.

### CODE REVIEW

File: sim-lib/src/lib.rs [hunk 1]
```diff
@@ -233,6 +233,7 @@ impl Simulation {
             self.activity.len(),
             self.nodes.len()
         );
+        let mut tasks = JoinSet::new();
```
Comment (carlaKC) [inline sim-lib/src/lib.rs:245]: noice 🥇

File: sim-cli/src/main.rs [hunk 2]
```diff
@@ -16,19 +16,35 @@ struct Cli {
     config: PathBuf,
     #[clap(long, short)]
     total_time: Option<u32>,
+    #[clap(long, short)]
+    debug: bool,
```
Comment (carlaKC) [inline sim-cli/src/main.rs:20]: Bool config option vs passing in a string that allows enabling other values like `trace`?

**Review by carlaKC — APPROVED** [review]
LGTM - possibly want to update log config to something like `--log-level=value` so that we can set any logging level but already a big improvement!

File: sim-cli/src/main.rs [hunk 3]
```diff
@@ -14,19 +14,37 @@ use simple_logger::SimpleLogger;
...
     config: PathBuf,
+    #[clap(long, short)]
+    total_time: Option<u32>,
+    #[clap(long, short)]
+    debug: bool,
 }
 
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
+    let cli = Cli::parse();
```
Comment (okjodom) [inline sim-cli/src/main.rs:25]: would we also want to read in `RUST_LOG` environment at this point?

**Review by okjodom — APPROVED** [review]

File: sim-cli/src/main.rs (hunk 3 above)
Comment (sr-gi) [reply sim-cli/src/main.rs:25]: I don't think I follow

**Pull request merged**